test/lint/lint-includes.sh

```diff
@@ -61,6 +61,7 @@ EXPECTED_BOOST_INCLUDES=(
     boost/multi_index/sequenced_index.hpp
     boost/multi_index_container.hpp
     boost/optional.hpp
+    boost/optional/optional_io.hpp
```

Comment:
Is there a C++17 equivalent for `optional_io`?

Reply:
Unfortunately it doesn't look like it to me. [The contents of the boost header](https://www.boost.org/doc/libs/1_65_0/boost/optional/optional_io.hpp) don't seem to have any equivalent implementations [in c++17](https://en.cppreference.com/w/cpp/utility/optional), but that was a good idea to check. I tried removing the inclusion of this and it broke the build, so I guess it's required.

Reply:
Okay so you're not intentionally using anything from it, needing to include it is only a by-effect of how boost is structured internally?
Interesting but yes I guess no way around it then.

Reply:
You can avoid this by using `std::optional`s everywhere. Minimal diff to get this working here (but other uses of boost::optional also need to be removed):

```diff
diff --git a/src/test/validation_tests.cpp b/src/test/validation_tests.cpp
index 37450ec18f..be9b6ccad2 100644
--- a/src/test/validation_tests.cpp
+++ b/src/test/validation_tests.cpp
@@ -5,14 +5,14 @@
 #include <chainparams.h>
 #include <net.h>
 #include <signet.h>
-#include <optional.h>
 #include <uint256.h>
 #include <validation.h>
 
 #include <test/util/setup_common.h>
 
 #include <boost/test/unit_test.hpp>
-#include <boost/optional/optional_io.hpp>
+
+#include <optional>
 
 BOOST_FIXTURE_TEST_SUITE(validation_tests, TestingSetup)
 
@@ -133,14 +133,14 @@ BOOST_AUTO_TEST_CASE(test_assumeutxo)
 
     for (auto empty : bad_heights) {
         const auto out = ExpectedAssumeutxo(empty, *params);
-        BOOST_CHECK_EQUAL(out, nullopt);
+        BOOST_CHECK(!out.has_value());
     }
 
-    const auto out110 = ExpectedAssumeutxo(110, *params).get();
+    const auto out110 = ExpectedAssumeutxo(110, *params).value();
     BOOST_CHECK_EQUAL(out110.hash_serialized, uint256S("76fd7334ac7c1baf57ddc0c626f073a655a35d98a4258cd1382c8cc2b8392e10"));
     BOOST_CHECK_EQUAL(out110.nChainTx, (unsigned int)110);
 
-    const auto out210 = ExpectedAssumeutxo(210, *params).get();
+    const auto out210 = ExpectedAssumeutxo(210, *params).value();
     BOOST_CHECK_EQUAL(out210.hash_serialized, uint256S("9c5ed99ef98544b34f8920b6d1802f72ac28ae6e2bd2bd4c316ff10c230df3f2"));
     BOOST_CHECK_EQUAL(out210.nChainTx, (unsigned int)210);
 }
diff --git a/src/validation.cpp b/src/validation.cpp
index e25d38d89f..09621420b5 100644
--- a/src/validation.cpp
+++ b/src/validation.cpp
@@ -5185,7 +5185,7 @@ CChainState& ChainstateManager::InitializeChainstate(CTxMemPool& mempool, const
     return *to_modify;
 }
 
-Optional<AssumeutxoData> ExpectedAssumeutxo(const int height, const CChainParams& chainparams)
+std::optional<AssumeutxoData> ExpectedAssumeutxo(const int height, const CChainParams& chainparams)
 {
     const MapAssumeutxo& valid_assumeutxos_map = chainparams.Assumeutxo();
     const auto assumeutxo_found = valid_assumeutxos_map.find(height);
diff --git a/src/validation.h b/src/validation.h
index 0b8a72b8b0..db2cc1b643 100644
--- a/src/validation.h
+++ b/src/validation.h
@@ -29,6 +29,7 @@
 #include <atomic>
 #include <map>
 #include <memory>
+#include <optional>
 #include <set>
 #include <stdint.h>
 #include <string>
@@ -1011,6 +1012,6 @@ inline bool IsBlockPruned(const CBlockIndex* pblockindex)
  *
  * @returns empty if no assumeutxo configuration exists for the given height.
  */
-Optional<AssumeutxoData> ExpectedAssumeutxo(const int height, const CChainParams& params);
+std::optional<AssumeutxoData> ExpectedAssumeutxo(const int height, const CChainParams& params);
 
 #endif // BITCOIN_VALIDATION_H
```

We really shouldn't be adding new boost dependencies into validation.h, so I think changing all the `boost::optional`s to `std::optional`s needs to happen anyway.

Reply:
boost::optional -> std::optional has now been merged: #20671.

This boost include can be removed (and some of the optional methods need to be changed).